### DISCUSSION TITLE
Welcome to P3 Model Discussions!

### DISCUSSION THREAD

**Discussion by technites-pl** [P3-model/P3-model#4] [Announcements]
## 👋 Welcome!
  We’re using Discussions as a place to connect with other members of our community. We hope that you:
  * Ask questions you’re wondering about.
  * Share ideas.
  * Engage with other community members.
  * Welcome others and are open-minded. Remember that this is a community we
  build together 💪.

  To get started, comment below with an introduction of yourself and tell us about what you do with this community.

**Comment by sculpt0r** [comment]
Hi,
I get here from bettersoftwaredesign.pl [episode 60](https://bettersoftwaredesign.pl/episodes/60).

I would like to follow the idea - since lack of proper 'zoomable` documentation is really painful. I would like to contribute around JS/TS tooling, but C# is also doable :)

My first concern: would you like to create some 'open stadard' format which will be produced by the parsers. Then this format will be available by some future one single tool able to read this standard?

**Reply by technites-pl** [reply]
Hi,
thanks for your interest!

We are going to create open standard that can be used by any tool. Parsers for each platform will use this standard as an output format. As a next step we're also going to provide visualization tool. Of course other third party visualization tools can use this format. Parsers (and all other parts of P3) will be designed in extensibility in mind so other output formats will be ease to add.

In a week or two I'll add POC version of C# parser. Instead of standardized output and independent visualization tool I'll use predefined set of markdown pages with Mermaid diagrams ( = another output format mentioned above). The main goal for this POC is to visualize the idea on working example without investing in complex visualization tool.

Porting C# parser to TS/JS will be great option.

**Reply by sculpt0r** [reply]
Great to hear that :) So I'll follow changes, maybe try to look at the C# part of the code :)

**Comment by manhunto** [comment]
Hi, 
I saw your presentation on boiling frogs and listened to your talk with Mariusz Gil on better software design, and I'm amazed by this idea. I can contribute by writing a parser for PHP :)

**Reply by technites-pl** [reply]
Hi,
great to hear that!

As I mentioned above POC of C# parser will be ready in a week or two. It'll be great time to join forces, review design and start porting it to PHP and other platforms.

**Comment by koksyn** [comment]
Hello, 
I didn't have possibility to see your presentation, but I listened the Mariusz Gil BSD podcast with you and it's really encouraged me to contribute as well. I offer my help with porting annotations, parser and other tools for Java/Kotlin environments.

**Reply by technites-pl** [reply]
Great! As soon as I have POC ready we'll schedule a meeting to plan next steps.

**Comment by technites-pl** [comment]
Hi everyone,

Initial version of Parser POC for C# is ready, check https://github.com/P3-model/P3-model-dotnet
It's tested only on Ubuntu and .net 6!
All tests are done on https://github.com/itlibrium/DDD-starter-dotnet. This repository contains enough P3 Model Annotations and other metadata files to play with the Parser POC in it's current version.
Output is in simple Markdown files and Mermaid diagrams for now but it's only to speed up experiments.

**Reply by technites-pl** [reply]
@sculpt0r @manhunto @koksyn @MaciejTrojniarz lets schedule a kick-off meeting (zoom): https://doodle.com/meeting/participate/id/dwKYg1md
Please vote by the end of the week.

**Reply by technites-pl** [reply]
According to Doodle, the best time to meet is 14.06 8:00-9:00.
We'll paste zoom link here an hour before the meeting.

**Reply by technites-pl** [reply]
Link to today's meeting: https://us02web.zoom.us/j/87361545574?pwd=NDJLSjViaGRwVEE1SGFEaGtkY0dzQT09

**Comment by MaciejTrojniarz** [comment]
If you need someone to work on JAVA version I'm happy to help.

**Reply by technites-pl** [reply]
Sure we need ;)
